2.0/m_opban.cpp

```diff
@@ -12,51 +12,63 @@
 /* $ModAuthorMail: sebastian@sebbe.eu */
 /* $ModDesc: Implements extban +b o: Prevents these persons from aquiring a privileged position */
 /* $ModDepends: core 2.0 */
+/* $ModConfig: Optional: <opban requiredrank="50000"> sets required rank to set/unset +b o, defaults to q */
 
 #include "inspircd.h"
 class ModuleOPBan : public Module
 {
-public:
+ private:
+	unsigned int RequiredRank;
+ public:
 	void init()
 	{
-		Implementation eventlist[] = { I_OnRawMode, I_On005Numeric };
+		Implementation eventlist[] = { I_OnRawMode, I_On005Numeric, I_OnRehash };
 		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
 	}
-  
 	Version GetVersion()
 	{
 		return Version("Implements extban +b o: Prevents these persons from aquiring a privileged position",VF_OPTCOMMON);
 	}
 
+	void OnRehash(User* user)
+	{
+		this->RequiredRank = 50000;
+		ConfigTag* conftag = ServerInstance->Config->ConfValue("opban");
+		if (conftag)
```

Comment:
There's no reason for this if statement. If theres no config tag an empty tag will be returned.

Reply:
aah, tought it was going to return null, which would result in a runtime error and a crash if I would try to GetInt() on a null value. Fixing.